packages/build/src/install/missing.js

```diff
@@ -21,16 +23,56 @@ export const installMissingPlugins = async function ({ missingPlugins, autoPlugi
   await addExactDependencies({ packageRoot: autoPluginsDir, isLocal: mode !== 'buildbot', packages })
 }
 
-export const installIntegrationPlugins = async function ({ integrations, autoPluginsDir, mode, logs }) {
-  const packages = integrations.map(getIntegrationPackage)
-  logInstallIntegrations(logs, integrations)
+export const installIntegrationPlugins = async function ({ integrations, autoPluginsDir, mode, logs, context }) {
+  const integrationsToBuild = integrations.filter(
+    (integration) => typeof integration.dev !== 'undefined' && context === 'dev',
+  )
+  if (integrationsToBuild.length) {
+    logSubHeader(logs, 'Building integrations')
+    logArray(
+      logs,
+      integrationsToBuild.map(({ slug, dev: { path } }) => `${slug} from ${path}`),
+    )
+  }
+  const packages = (
+    await Promise.all(integrations.map((integration) => getIntegrationPackage({ integration, context })))
+  ).filter(Boolean)
+  logInstallIntegrations(
+    logs,
+    integrations.filter((integration) =>
+      integrationsToBuild.every((compiledIntegration) => integration.slug !== compiledIntegration.slug),
+    ),
+  )
 
   await createAutoPluginsDir(logs, autoPluginsDir)
   await addExactDependencies({ packageRoot: autoPluginsDir, isLocal: mode !== 'buildbot', packages })
 }
 
-const getIntegrationPackage = function ({ version }) {
-  return `${version}/packages/buildhooks.tgz`
+const getIntegrationPackage = async function ({ integration: { version, dev }, context }) {
+  if (typeof version !== 'undefined') {
+    return `${version}/packages/buildhooks.tgz`
+  }
+
+  if (typeof dev !== 'undefined' && context === 'dev') {
+    const { path } = dev
+
+    const integrationDir = resolve(process.cwd(), path)
```

Comment:
nit: `process.cwd()` is the default "from" of `path.resolve`, so you don't need to supply it here.

```suggestion
    const integrationDir = resolve(path)
```